### PR TITLE
Push only essential directories before `login`

### LINKED ISSUE #2681: Pull `TMT_PLAN_DATA` at the end of the `finish` step
Use case: run script in `finish` which writes data to `$TMT_PLAN_DATA` and use `login` (`tmt run -a login`)

We need to 'pull' data before login 'push' or keep existing data on the guest. Currently the 'login' deletes everything produced by finish (as no pull happens before)

### PR DESCRIPTION
**Pull request by psss**
Do not attempt to push the whole workdir to the guest before the `login` action in order to prevent overwriting user data stored in the `TMT_PLAN_DATA` directory.

Fix #2681.

Pull Request Checklist

* [x] implement the feature
* [x] extend the test coverage
* [x] include a release note

### CODE REVIEW

File: tmt/steps/__init__.py [hunk 1]
```diff
@@ -2444,7 +2444,7 @@ def _login(
         for guest in self.parent.plan.provision.ready_guests:
             # Attempt to push the workdir to the guest
             try:
-                guest.push()
+                guest.push(self.parent.plan.worktree)
```
Comment (LecrisUT) [inline tmt/steps/__init__.py]: I believe this is already the default if we pass `None` isn't it? No complains on making it more explicit, but could we make it `source=..., destination=None` or such?

File: tests/login/plan-data/test.sh [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    rlPhaseStartSetup
+        rlRun "run=\$(mktemp -d)" 0 "Create a run directory"
+        rlRun "pushd data"
+        rlRun "PROVISION_HOW=${PROVISION_HOW:-virtual}"
+    rlPhaseEnd
+
+    rlPhaseStartTest
+        # Log in at the end of prepare, execute and finish
+        rlRun "tmt run --all --id $run provision --how $PROVISION_HOW \
+            login --step prepare --step execute --step finish --command 'ls -l ../data'"
```
Comment (LecrisUT) [inline tests/login/plan-data/test.sh:14]: Shouldn't we have a test that does only `provision` and `finish` to catch the issue mentioned in the comments?

File: tmt/steps/__init__.py (hunk 1 above)
Comment (psss) [reply tmt/steps/__init__.py]: By default the whole plan workdir is pushed:

https://github.com/teemtee/tmt/blob/518e15b46e0217869cc6013da5032ee1c60c3d6e/tmt/steps/provision/__init__.py#L2826-L2828

Here we want just the `plan_workdir / "tree"` to prevent overwriting `plan_workdir / "data"`.

Comment (LecrisUT) [reply tmt/steps/__init__.py]: Ah ok, makes more sense. But still let's add the `source` keyword explicitly, don't quite like the ambiguity of the current interface.

File: tests/login/plan-data/test.sh (hunk 2 above)
Comment (psss) [reply tests/login/plan-data/test.sh:14]: The problem was actually revealed thanks to `/tests/provision/virtual.testcloud` here:

https://github.com/teemtee/tmt/blob/518e15b46e0217869cc6013da5032ee1c60c3d6e/tests/provision/virtual.testcloud/test.sh#L51-L57

But it makes sense to explicitly cover that scenario here as well. Added in 785a6114.

Comment (LecrisUT) [reply tests/login/plan-data/test.sh:14]: Yeah, it is quite the edge-case so it's nice to have something more dedicated here.

File: tmt/steps/__init__.py (hunk 1 above)
Comment (psss) [reply tmt/steps/__init__.py]: Sure, added in a8a486b5.

**Review by vaibhavdaren — APPROVED** [review]
